main.go

```diff
@@ -104,22 +145,33 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	unique := sameNameOnDB(newBook) //Verify if the already there is a book with the same name in the database
+	unique, err := sameNameOnDB(newBook) //Verify if the already there is a book with the same name in the database
+	if err != nil {
+		log.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
 	if !unique {
 		responseJSON(w, http.StatusBadRequest, errResponseCreateBookNameConflict)
 		return
 	}
 
 	newBook.ID = uuid.New() //Atribute an ID to the entry
 
-	fail, storedBook := storeOnDB(newBook) //Store the book in the database
-	if fail {
+	storedBook, err := storeOnDB(newBook) //Store the book in the database
+	if err != nil {
+		if errors.Is(err, errBookNotFound) {
+			log.Println(err)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
```

Comment:
Does this makes sense? Returning a `not found` when failing to create a book.
You are not passing no ID in the URL, returning `404` to the user states that the path does not exist, but it does, the service only had issues inserting a new book.
So, I would say, lets not care about which error occurred during insertion for now. We may need to check it in the future to make sure we didn't got any conflicts.

Reply:
I knew it! I took a while staring at this point when I was writting it. I was wondering that, if I chose 500 instead of 404, checking which error were coming back was almost useless, except for logging it, so I choose 404. Now I see better that it was a mistake. Is it a 500, right?

Reply:
In this case yes, `not found` doesn't make much sense in an insertion. Remove this `error.Is` check altogether. Let it fall into the 500 case.